src/equisolve/numpy/models/linear_model.py

```diff
@@ -108,7 +112,10 @@ def _validate_params(
             if sample_weight is not None:
                 sw_block = sample_weight.block(key)
                 _check_blocks(
-                    X_block, sw_block, props=["samples"], fname="_validate_params"
+                    X_block,
+                    sw_block,
+                    props=["samples", "components"],
```

Comment:
I think sample weights should not be allowed to have components (it does not make sense to have different weights for different components)

Reply:
Are you okay with UserWarning? I agree with what you say, but sometimes one does not foresee how the user uses a library and for these cases you block the user by making it restrictive by raising an Error. I don't have a strong opinion here.

Reply:
I tend to err on the side of caution and disable more things. It is much easier from a library evolution point of view to add new functionality rather than remove it, and we don't need/use a part of the code, it is very easy for it to contains bugs/have bugs introduced when we update the code surrounding it.

Reply:
I am fine with not allowing components. However, if we do not allow components we have to extend the sample_weights so that the shape of the array the same as for X and Y. This is a bit more complicated